src/components/admin.tsx

```diff
@@ -420,12 +425,19 @@ function Admin({ user }: adminProps) {
   };
 
   const setTimedLink = (
+    linktype: number,
+    postalcode: String,
     addressLinkId: String,
     hours = DEFAULT_SELF_DESTRUCT_HOURS
   ) => {
-    return pollingFunction(() =>
-      set(ref(database, `links/${addressLinkId}`), addHours(hours))
-    );
+    const link = new LinkSession();
+    link.tokenEndtime = addHours(hours);
+    link.postalCode = postalcode as string;
+    link.linkType = linktype;
+    return pollingFunction(async () => {
+      set(ref(database, `links/${addressLinkId}`), link);
+      await triggerPostalCodeListeners(link.postalCode);
```

Comment:
![IMG_525E32A2D95D-1](https://user-images.githubusercontent.com/40650158/206979618-daa7b0cb-00ee-40e1-811c-8a8b656c697d.jpeg)

This registers the link as a count even when I abort the share action for some reason. Perhaps we can add some sort of status to the LinkSession ? On successfully share, set it "active". If cancelled, then "inactive". Something along those lines.

Reply:
I don't think the share action can be cancelled because once activated it is out of our hands

Reply:
Fixed by creating link after user selects app to share.